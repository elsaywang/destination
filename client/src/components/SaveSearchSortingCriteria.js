import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import RadioGroup from '@react/react-spectrum/RadioGroup';
import Radio from '@react/react-spectrum/Radio';
import Select from '@react/react-spectrum/Select';

function SaveSearchSortingCriteria({
    selectOptions,
    radioGroupOptions,
    onDefaultSortingChange,
    handleRadioChange,
    radioCheckedValue,
    radioGroupStyle,
}) {
    return (
        <Fragment>
            <Select options={selectOptions} onChange={onDefaultSortingChange} />
            <div className={radioGroupStyle}>
                <RadioGroup>
                    {radioGroupOptions.map(({ label, value }) => (
                        <Radio
                            key={value}
                            {...{ label, value }}
                            checked={value === radioCheckedValue}
                            onChange={handleRadioChange}
                        />
                    ))}
                </RadioGroup>
            </div>
        </Fragment>
    );
}

SaveSearchSortingCriteria.propTypes = {
    selectOptions: PropTypes.array,
    radioGroupOptions: PropTypes.array,
    onDefaultSortingChange: PropTypes.func,
    handleRadioChange: PropTypes.func,
    radioCheckedValue: PropTypes.string,
    radioGroupStyle: PropTypes.string,
};

export default SaveSearchSortingCriteria;
