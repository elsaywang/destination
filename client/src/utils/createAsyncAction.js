function createAsyncAction(type, fn) {
    return (...args) => async dispatch => {
        let response;

        try {
            response = await fn(...args);

            if (response.ok) {
                const json = await response.json();

                dispatch({
                    type: `${type}`,
                    payload: json,
                });
            } else {
                throw new Error(response.statusText);
            }
        } catch (error) {
            dispatch({
                type: `${type}_REJECTED`,
                error: true,
                payload: error,
            });
        }

        return response;
    };
}

export { createAsyncAction };
