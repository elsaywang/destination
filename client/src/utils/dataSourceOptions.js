export const formatDataSourceLabel = (dataSourceId, name) => `${name} (${dataSourceId})`;

export const isValidDataSourceId = (dataSources, selectedDataSourceId) =>
    dataSources.filter(({ dataSourceId }) => dataSourceId === selectedDataSourceId).length;
