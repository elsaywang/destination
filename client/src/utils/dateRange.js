import moment from 'moment';
import {
    customDateFormat,
    defaultDateRangeDays,
    defaultDateRangePreset,
} from '../constants/dateRangeConstants';

export const getNow = () => moment.utc();

export const getDefaultCustomStartDate = () =>
    getNow()
        .subtract(defaultDateRangeDays, 'days')
        .format(customDateFormat);

export const getDefaultCustomEndDate = () => getNow().format(customDateFormat);
