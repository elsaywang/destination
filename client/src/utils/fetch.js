import { getCsrfToken } from '../lib/getCsrfToken';

export const getOptionsWithAAMAuth = (options = { headers: {} }) => ({
    ...options,
    headers: {
        'content-type': 'application/json',
        ...options.headers,
        'AAM-CSRF-Token': getCsrfToken(),
    },
    credentials: 'same-origin',
});

export default function(url, options) {
    return fetch(url, getOptionsWithAAMAuth(options));
}
