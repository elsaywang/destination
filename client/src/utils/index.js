export const fetchTraits = sids => {
    // fetch(`/api/traits/${sids}`
    return fetch('/api/traits/');
};
