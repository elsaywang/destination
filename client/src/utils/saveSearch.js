import { allSignalsColumns } from '../constants/columns';

export const saveThisSearch = 'Save This Search';
export const checkBoxLabel = 'Track this search result in dashboard';
export const defaultSorting = 'Default Sorting';
export const textFieldLabelName = 'Name';
export const textFieldPlaceHolder = 'Enter a name for this saved search';

export const sortingOptions = allSignalsColumns.map(({ title, key }) => ({
    label: title,
    value: key,
}));

export const radioGroupOptions = [
    { label: 'Descending', value: 'descending' },
    { label: 'Ascending', value: 'ascending' },
];
