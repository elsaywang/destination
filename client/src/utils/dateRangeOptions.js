import { validDateRangeDays } from '../constants/dateRangeConstants';
import { defaultMaxSignalRetentionDays } from '../constants/limitConstants';
import { dateRangeDaysToPreset } from './dateRange';

export const getDateRangesWithinRetentionPolicy = (
    maxSignalRetentionDays = defaultMaxSignalRetentionDays,
) => validDateRangeDays.filter(dateRangeDays => maxSignalRetentionDays >= dateRangeDays);

export const getDateRangeOptions = (dateRangeDays = validDateRangeDays) => [
    ...dateRangeDays.map(days => ({
        label: `Last ${days} Day${days === 1 ? '' : 's'}`,
        value: dateRangeDaysToPreset(days),
    })),
    {
        label: 'Custom Date Range',
        value: 'custom',
    },
];

export const getDateRangeOptionsWithinRetentionPolicy = (
    maxSignalRetentionDays = defaultMaxSignalRetentionDays,
) => getDateRangeOptions(getDateRangesWithinRetentionPolicy(maxSignalRetentionDays));

export const getDateRangeLabel = dateRangePreset => {
    const dateRangeOption = getDateRangeOptions().find(option => option.value === dateRangePreset);

    return dateRangeOption ? dateRangeOption.label : '';
};
