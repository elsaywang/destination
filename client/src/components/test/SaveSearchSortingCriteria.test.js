import React from 'react';
import { shallow } from 'enzyme';
import SaveSearchSortingCriteria from '../SaveSearchSortingCriteria';
import RadioGroup from '@react/react-spectrum/RadioGroup';
import Radio from '@react/react-spectrum/Radio';
import Select from '@react/react-spectrum/Select';
import styles from '../SaveSearchExecution.css';
import { defaultSorting, radioGroupOptions, sortingOptions } from '../../utils/saveSearch';

describe('when CheckBox is checked', () => {
    const mockFn = jest.fn();
    const wrapper = shallow(
        <SaveSearchSortingCriteria
            selectOptions={sortingOptions}
            radioGroupOptions={radioGroupOptions}
            onDefaultSortingChange={mockFn}
            handleRadioChange={mockFn}
            radioCheckedValue="descending"
            radioGroupStyle={styles.contentRadioGroup}
        />,
    );
    it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('renders <Label/> with correct label/value and children of <Select/> and <RadioGroup/> including 2 <Radio/>', () => {
        expect(wrapper.find(Select)).toBeTruthy();
        expect(wrapper.find(RadioGroup).exists()).toBeTruthy();
        expect(wrapper.find(Radio).length).toBe(2);
    });

    it('renders <Select/> with correct option list', () => {
        expect(wrapper.find(Select).props().options.length).toBe(sortingOptions.length);
        expect(wrapper.find(Select).props().options).toBe(sortingOptions);
    });

    it(`renders <RadioGroup/> including 2 <Radio/> with correct label and value`, () => {
        radioGroupOptions.map(({ label, value }, index) => {
            expect(
                wrapper
                    .find(Radio)
                    .at(index)
                    .props().label,
            ).toEqual(label);
            expect(
                wrapper
                    .find(Radio)
                    .at(index)
                    .props().value,
            ).toEqual(value);
            expect(
                wrapper
                    .find(Radio)
                    .at(index)
                    .props().checked,
            ).toEqual(value === 'descending');
            expect(
                wrapper
                    .find(Radio)
                    .at(index)
                    .props().radioGroupStyle,
            ).toEqual(styles.radioGroupStyle);
        });
    });
});
