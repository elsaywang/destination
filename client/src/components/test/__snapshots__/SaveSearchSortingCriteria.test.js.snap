// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when CheckBox is checked matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SaveSearchSortingCriteria
    handleRadioChange={[MockFunction]}
    onDefaultSortingChange={[MockFunction]}
    radioCheckedValue="descending"
    radioGroupOptions={
      Array [
        Object {
          "label": "Descending",
          "value": "descending",
        },
        Object {
          "label": "Ascending",
          "value": "ascending",
        },
      ]
    }
    radioGroupStyle={undefined}
    selectOptions={
      Array [
        Object {
          "label": "Key-Value Pair",
          "value": "keyValuePairs",
        },
        Object {
          "label": "Signal Type",
          "value": "signalType",
        },
        Object {
          "label": "Signal Source",
          "value": "signalSource",
        },
        Object {
          "label": "Total Counts",
          "value": "totalCounts",
        },
        Object {
          "label": "Percentage Change",
          "value": "percentageChange",
        },
        Object {
          "label": "Included In Traits",
          "value": "includedInTraits",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Select
          onChange={[MockFunction]}
          options={
            Array [
              Object {
                "label": "Key-Value Pair",
                "value": "keyValuePairs",
              },
              Object {
                "label": "Signal Type",
                "value": "signalType",
              },
              Object {
                "label": "Signal Source",
                "value": "signalSource",
              },
              Object {
                "label": "Total Counts",
                "value": "totalCounts",
              },
              Object {
                "label": "Percentage Change",
                "value": "percentageChange",
              },
              Object {
                "label": "Included In Traits",
                "value": "includedInTraits",
              },
            ]
          }
        />,
        <div
          className={undefined}
        >
          <RadioGroup
            labelsBelow={false}
            onChange={[Function]}
            vertical={false}
          >
            <Radio
              checked={true}
              label="Descending"
              onChange={[MockFunction]}
              value="descending"
            />
            <Radio
              checked={false}
              label="Ascending"
              onChange={[MockFunction]}
              value="ascending"
            />
          </RadioGroup>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onChange": [MockFunction],
          "options": Array [
            Object {
              "label": "Key-Value Pair",
              "value": "keyValuePairs",
            },
            Object {
              "label": "Signal Type",
              "value": "signalType",
            },
            Object {
              "label": "Signal Source",
              "value": "signalSource",
            },
            Object {
              "label": "Total Counts",
              "value": "totalCounts",
            },
            Object {
              "label": "Percentage Change",
              "value": "percentageChange",
            },
            Object {
              "label": "Included In Traits",
              "value": "includedInTraits",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <RadioGroup
            labelsBelow={false}
            onChange={[Function]}
            vertical={false}
          >
            <Radio
              checked={true}
              label="Descending"
              onChange={[MockFunction]}
              value="descending"
            />
            <Radio
              checked={false}
              label="Ascending"
              onChange={[MockFunction]}
              value="ascending"
            />
          </RadioGroup>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Radio
                checked={true}
                label="Descending"
                onChange={[MockFunction]}
                value="descending"
              />,
              <Radio
                checked={false}
                label="Ascending"
                onChange={[MockFunction]}
                value="ascending"
              />,
            ],
            "labelsBelow": false,
            "onChange": [Function],
            "vertical": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "descending",
              "nodeType": "function",
              "props": Object {
                "checked": true,
                "label": "Descending",
                "onChange": [MockFunction],
                "value": "descending",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "ascending",
              "nodeType": "function",
              "props": Object {
                "checked": false,
                "label": "Ascending",
                "onChange": [MockFunction],
                "value": "ascending",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Select
            onChange={[MockFunction]}
            options={
              Array [
                Object {
                  "label": "Key-Value Pair",
                  "value": "keyValuePairs",
                },
                Object {
                  "label": "Signal Type",
                  "value": "signalType",
                },
                Object {
                  "label": "Signal Source",
                  "value": "signalSource",
                },
                Object {
                  "label": "Total Counts",
                  "value": "totalCounts",
                },
                Object {
                  "label": "Percentage Change",
                  "value": "percentageChange",
                },
                Object {
                  "label": "Included In Traits",
                  "value": "includedInTraits",
                },
              ]
            }
          />,
          <div
            className={undefined}
          >
            <RadioGroup
              labelsBelow={false}
              onChange={[Function]}
              vertical={false}
            >
              <Radio
                checked={true}
                label="Descending"
                onChange={[MockFunction]}
                value="descending"
              />
              <Radio
                checked={false}
                label="Ascending"
                onChange={[MockFunction]}
                value="ascending"
              />
            </RadioGroup>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onChange": [MockFunction],
            "options": Array [
              Object {
                "label": "Key-Value Pair",
                "value": "keyValuePairs",
              },
              Object {
                "label": "Signal Type",
                "value": "signalType",
              },
              Object {
                "label": "Signal Source",
                "value": "signalSource",
              },
              Object {
                "label": "Total Counts",
                "value": "totalCounts",
              },
              Object {
                "label": "Percentage Change",
                "value": "percentageChange",
              },
              Object {
                "label": "Included In Traits",
                "value": "includedInTraits",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <RadioGroup
              labelsBelow={false}
              onChange={[Function]}
              vertical={false}
            >
              <Radio
                checked={true}
                label="Descending"
                onChange={[MockFunction]}
                value="descending"
              />
              <Radio
                checked={false}
                label="Ascending"
                onChange={[MockFunction]}
                value="ascending"
              />
            </RadioGroup>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Radio
                  checked={true}
                  label="Descending"
                  onChange={[MockFunction]}
                  value="descending"
                />,
                <Radio
                  checked={false}
                  label="Ascending"
                  onChange={[MockFunction]}
                  value="ascending"
                />,
              ],
              "labelsBelow": false,
              "onChange": [Function],
              "vertical": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "descending",
                "nodeType": "function",
                "props": Object {
                  "checked": true,
                  "label": "Descending",
                  "onChange": [MockFunction],
                  "value": "descending",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "ascending",
                "nodeType": "function",
                "props": Object {
                  "checked": false,
                  "label": "Ascending",
                  "onChange": [MockFunction],
                  "value": "ascending",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
